api/addons/serializers.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import serializers as ser
+from api.base.serializers import JSONAPISerializer, LinksField
+from api.base.utils import absolute_reverse
+
+class NodeAddonFolderSerializer(JSONAPISerializer):
+    """
+    Overrides AddonSettingsSerializer to return node-specific fields
+    """
+    class Meta:
+        type_ = 'node_addon_folders'
+
+    id = ser.CharField(source='provider', read_only=True)
+    kind = ser.CharField(default='folder', read_only=True)
+    name = ser.CharField(read_only=True)
+    path = ser.CharField(read_only=True)
+    folder_id = ser.CharField(read_only=True)
+
+    links = LinksField({
+        'children': 'get_child_folders'
+    })
+
+    def get_child_folders(self, obj):
```

Comment:
For some reason I'm thought `obj` was a class instance, but later it's being accessed like a <dict>. Is this correct? Should we be using `getattr` instead?